test/Faker/Provider/InternetTest.php

```diff
@@ -40,29 +40,65 @@ public function localeDataProvider()
     /**
      * @link http://stackoverflow.com/questions/12026842/how-to-validate-an-email-address-in-php
      *
-     * @requires PHP 5.4
      * @dataProvider localeDataProvider
      */
     public function testEmailIsValid($locale)
     {
+        if ($locale !== 'en_US' && !class_exists('Transliterator')) {
+            $this->markTestSkipped('Transliterator class not available (intl extension)');
+        }
+
         $this->loadLocalProviders($locale);
         $pattern = '/^(?!(?:(?:\\x22?\\x5C[\\x00-\\x7E]\\x22?)|(?:\\x22?[^\\x5C\\x22]\\x22?)){255,})(?!(?:(?:\\x22?\\x5C[\\x00-\\x7E]\\x22?)|(?:\\x22?[^\\x5C\\x22]\\x22?)){65,}@)(?:(?:[\\x21\\x23-\\x27\\x2A\\x2B\\x2D\\x2F-\\x39\\x3D\\x3F\\x5E-\\x7E]+)|(?:\\x22(?:[\\x01-\\x08\\x0B\\x0C\\x0E-\\x1F\\x21\\x23-\\x5B\\x5D-\\x7F]|(?:\\x5C[\\x00-\\x7F]))*\\x22))(?:\\.(?:(?:[\\x21\\x23-\\x27\\x2A\\x2B\\x2D\\x2F-\\x39\\x3D\\x3F\\x5E-\\x7E]+)|(?:\\x22(?:[\\x01-\\x08\\x0B\\x0C\\x0E-\\x1F\\x21\\x23-\\x5B\\x5D-\\x7F]|(?:\\x5C[\\x00-\\x7F]))*\\x22)))*@(?:(?:(?!.*[^.]{64,})(?:(?:(?:xn--)?[a-z0-9]+(?:-+[a-z0-9]+)*\\.){1,126}){1,}(?:(?:[a-z][a-z0-9]*)|(?:(?:xn--)[a-z0-9]+))(?:-+[a-z0-9]+)*)|(?:\\[(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){7})|(?:(?!(?:.*[a-f0-9][:\\]]){7,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?)))|(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){5}:)|(?:(?!(?:.*[a-f0-9]:){5,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3}:)?)))?(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))(?:\\.(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))){3}))\\]))$/iD';
         $emailAddress = $this->faker->email();
         $this->assertRegExp($pattern, $emailAddress);
     }
     
     /**
-     * @requires PHP 5.4
      * @dataProvider localeDataProvider
      */
     public function testUsernameIsValid($locale)
     {
+        if ($locale !== 'en_US' && !class_exists('Transliterator')) {
+            $this->markTestSkipped('Transliterator class not available (intl extension)');
+        }
+
         $this->loadLocalProviders($locale);
-        $pattern = '/^[A-Za-z0-9._]+$/';
+        $pattern = '/^(?![._])[A-Za-z0-9._]+(?<![._])$/';
```

Comment:
PR has been updated with this alternative, which I think is clearer. It's also stricter (forbid consecutive separators).